### ISSUE TITLE
Joined tables are automatically included in a query even if there are no fields selected from the table

### ISSUE THREAD

**Issue by TuringLovesDeathMetal** [lightdash/lightdash#753]
### Context
By default, we join to all tables specified in the meta joins even if no fields are selected from them. 

### Goal
Optimize the join behaviour in Lightdash by only including a join if a field from the joined table has been included in the query.

E.g. No fields have ben selected in this, but there is the join included in the query 

https://demo.lightdash.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/tables

![image](https://user-images.githubusercontent.com/31848148/140728630-32c6b890-ff0b-451c-b8df-c3c86e33b280.png)

**Comment by owlas** [comment]
I believe this is intentional in case there is a many-to-one join between the views in the explore. 

For example if the base table in user and I join to user-daily (which is many user-dailies to one user), then my table is actually one row per user-daily not one row per user.

If we don't always join then these scenarios query different tables:
1. Only using metrics from the base table "user" -> you would just be querying one-user-per-row
2. Using metrics from base table "user" and joined table "user-daily" -> you would be querying one-user-daily-per-row

**Comment by TuringLovesDeathMetal** [comment]
oooooh, I would not have expected that behaviour. 

Eg. if I had a metric in the `users` table called `total_purchase_amount` in the scenario above, I would _**not**_ expect it to be duplicated over a bunch of dates, then summed - I would only expect this duplication if I'd added something from `users_daily` (since then, I'd see the rows I'm duplicating over)

E.g. if I was looking at `total_purchase_amount`, which was a metric in `users`, grouped by postcode, then I would expect example #1. If I  then added **_any field_** from `user_stats`, I would expect something like example #2.

**example 1:**
```
SELECT
postcode,
SUM(purchase_amount) AS total_purchase_amount

FROM users
GROUP BY 1
```

**example 2:**
```
SELECT
users.postcode,
users_daily.date,
SUM(users.purchase_amount) AS total_purchase_amount

FROM users
LEFT JOIN users_daily USING(user_id)

GROUP BY 1, 2
```

**Comment by ZeRego** [comment]
@TuringLovesDeathMetal @owlas 
Have you have a look at this [looker article](https://help.looker.com/hc/en-us/articles/360023722974-A-Simple-Explanation-of-Symmetric-Aggregates-or-Why-On-Earth-Does-My-SQL-Look-Like-That-) ? 
Instead of having different behaviour/values depending on the selected dimension/metric selected they generate a massive query that figures out the correct value.

**Comment by TuringLovesDeathMetal** [comment]
ya, I had a look at this...it's actually quite complicated what they're doing behind the scenes (but really slick). 

How hard would it be to implement this?

**Comment by TuringLovesDeathMetal** [comment]
Closed because this has been solved in #905 

There is a bigger discussion around what we want the join logic to be happening here: https://github.com/lightdash/lightdash/issues/934
